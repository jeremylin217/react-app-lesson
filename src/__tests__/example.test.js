describe('test example 1', () => {
  it('two plus two is four', () => {
    expect(2 + 2).toBe(4)
    expect(2 + 2).toMatchSnapshot()
  })

  it('object assignment', () => {
    const data = { one: 1 }
    data['two'] = 2
    expect(data).toEqual({ one: 1, two: 2 })
  })

  it('null', () => {
    const n = null
    expect(n).toBeDefined()
  })

  const shoppingList = [
    'diapers',
    'kleenex',
    'trash bags',
    'paper towels',
    'beer',
  ]

  it('the shopping list has beer on it', () => {
    expect(shoppingList).toContain('beer')
  })
})

describe('test example 2', () => {
  it('call a mock function', () => {
    const mockFunc = jest.fn()
    mockFunc()
    expect(mockFunc).toBeCalled()
  })
})
