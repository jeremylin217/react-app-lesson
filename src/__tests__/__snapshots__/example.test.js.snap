// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test example 1 two plus two is four 1`] = `4`;
